ui/app/routes/vault/cluster/auth.js

```diff
@@ -7,27 +7,35 @@ import { service } from '@ember/service';
 import ClusterRouteBase from './cluster-route-base';
 import config from 'vault/config/environment';
 
-export default ClusterRouteBase.extend({
-  queryParams: {
-    authMethod: {
-      replace: true,
-    },
-  },
-  flashMessages: service(),
-  version: service(),
+export default class AuthRoute extends ClusterRouteBase {
+  queryParams = {
+    wrapped_token: { refreshModel: true },
+    authMethod: { replace: true },
+  };
+
+  @service api;
+  @service auth;
+  @service flashMessages;
+  @service version;
+
   beforeModel() {
-    return this._super().then(() => {
+    return super.beforeModel().then(() => {
       return this.version.fetchFeatures();
     });
-  },
-  model() {
-    return this._super(...arguments);
-  },
+  }
+
+  async model(params, transition) {
+    const wrapped_token = transition?.to?.queryParams?.wrapped_token;
+    if (wrapped_token) {
+      const clusterModel = this.modelFor('vault.cluster');
+      await this.unwrapToken(wrapped_token, clusterModel.id);
+    }
+    return super.model(...arguments);
+  }
 
   resetController(controller) {
-    controller.set('wrappedToken', '');
```

Comment:
resetController isn't hit since the model hook returns and logs a user in with a successful unwrap.